### PR TITLE
Rename `vec_unchop()` to `list_unchop()`

### LINKED ISSUE #1209: Possibly alias `vec_unchop()` with `list_unchop()`
It was unfortunate that we named this `vec_unchop()` rather than `list_unchop()` since the only allowed input is a list. Rather than removing it, we can alias it with `list_unchop()`.

We could also mark `vec_unchop()` with `@keywords internal`

### PR DESCRIPTION
**Pull request by DavisVaughan**
Closes #1209 

I went ahead and soft deprecated `vec_unchop()` as well. I think it is mostly us that use it, so having a soft deprecation warning show up in tests is probably a pretty low pain way to get us to move to `list_unchop()`

### CODE REVIEW

**Comment by lionel-** [comment]
Since we're now introducing a similar function in purrr for the name is still under discussion, maybe we should wait until this is resolved? Should we discuss the naming at the meeting?

**Comment by DavisVaughan** [comment]
I was under the impression that we'd have to do this renaming no matter what because `vec_unchop()` has the wrong prefix, but generally we want something that is symmetric with `vec_chop()` (so we'd keep the unchop suffix)

**Comment by DavisVaughan** [comment]
@lionel- do you want to do a review of this since we decided to push forward with this name change?

File: R/vctrs-deprecated.R [hunk 1]
```diff
@@ -114,3 +114,35 @@ vec_repeat <- function(x, each = 1L, times = 1L) {
...
+#' @inherit list_unchop return
+#'
+#' @keywords internal
+#' @export
+vec_unchop <- function(x,
+                       indices = NULL,
+                       ptype = NULL,
+                       name_spec = NULL,
+                       name_repair = c("minimal", "unique", "check_unique", "universal")) {
+  signal_soft_deprecated(paste_line(
```
[the remark this answers is not in the document]
Comment (DavisVaughan) [reply R/vctrs-deprecated.R:136]: So we should go ahead and `Import` lifecycle? Should the other deprecated functions also switch to using it?

[the remark this answers is not in the document]
Comment (lionel-) [reply R/vctrs-deprecated.R:136]: Yes but it's low priority to switch them.

**Pull request merged**